xdsl/dialects/func.py

```diff
@@ -20,6 +20,9 @@ class FuncOp(Operation):
     def verify_(self) -> None:
         # TODO: how to verify that there is a terminator?
         entry_block: Block = self.body.blocks[0]
+        # If this is an empty block (external function) then return
+        if len(entry_block.args) == 0 and len(entry_block.ops) == 0:
```

Comment:
IMO, we could also change the constructor s.t. it does not construct a block for external functions and then check that there is no block in the region. That would be cleaner, IMO.

I am not sure how MLIR looks internally, since a region with one empty block and a region without any blocks look the same in the printed IR, AFAIK.

Reply:
Are you Ok to do this as I think you have a clear idea of what it might look like?

Reply:
Sure, will open a PR on that :+1:

Reply:
It seems that MLIR indeed uses an empty block to encode empty regions. I am fine with this as is then.

([discussion](https://xdsl.zulipchat.com/#narrow/stream/362011-xDSL))